### PR TITLE
minio-cpp: add recipe

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **minio-cpp/0.3.0**

#### Motivation
The client library for minio (S3 compatible object storage).

#### Details
https://github.com/minio/minio-cpp/

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/minio-cpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("curlpp/0.8.1.cci.20240530", transitive_headers=True)
+        self.requires("inih/58")
+        self.requires("nlohmann_json/3.11.3", transitive_headers=True)
+        self.requires("openssl/3.3.2")
+        self.requires("pugixml/1.14")
+        self.requires("zlib/1.3.1")
```
Comment (AbrilRBS) [inline recipes/minio-cpp/all/conanfile.py:59]: Openssl and zlib should use version ranges!

File: recipes/minio-cpp/all/test_package/test_package.cpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#include <miniocpp/client.h>
+
+int main(int argc, char* argv[]) {
+  minio::s3::BaseUrl base_url("play.min.io");
+
+  minio::creds::StaticProvider provider(
+      "Q3AM3UQ867SPQQA43P2F", "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG");
```
Comment (AbrilRBS) [inline recipes/minio-cpp/all/test_package/test_package.cpp:7]: This reads a bit too much like IDs and connections, I'd simplify the test package a lot, a simple call would be ok, thanks!

**Review by AbrilRBS — COMMENTED** [review]
Thanks @toge! I have a few minor requests, otherwise looks good!

File: recipes/minio-cpp/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/minio-cpp/all/conanfile.py]: Thank you for your review.
This is the point I forgot to correct from my private recipe.
Fixed.

File: recipes/minio-cpp/all/test_package/test_package.cpp (hunk 2 above)
Comment (toge) [reply recipes/minio-cpp/all/test_package/test_package.cpp]: Simplified.

**Review by AbrilRBS — DISMISSED** [review]
Thanks!

File: recipes/minio-cpp/all/test_package/test_package.cpp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#include <miniocpp/client.h>
+
+int main(int argc, char* argv[]) {
+  minio::s3::BaseUrl base_url("example.com");
+
+  minio::creds::StaticProvider provider("access_key", "secret_key");
+
+  minio::s3::Client client(base_url, &provider);
```
Comment (AbrilRBS) [inline recipes/minio-cpp/all/test_package/test_package.cpp:8]: This only stores the attributes, connection is done thru method calls, 👍

**Review by uilianries — APPROVED** [review]
LGTM. I changed the CMakeDeps to customize the cmake targets, instead of patching the source.

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Pull request merged**